cpp/src/fil/fil.cu

```diff
@@ -535,15 +589,26 @@ int find_class_label_from_one_hot(L* vector, int len) {
 }
 
 template <typename fil_node_t, typename T, typename L>
-void tl2fil_leaf_payload(fil_node_t* fil_node, const tl::Tree<T, L>& tl_tree,
-                         int tl_node_id, const forest_params_t& forest_params) {
+void tl2fil_leaf_payload(fil_node_t* fil_node, int fil_node_id,
+                         const tl::Tree<T, L>& tl_tree, int tl_node_id,
+                         const forest_params_t& forest_params,
+                         std::vector<val_t>* vector_leaf) {
   auto vec = tl_tree.LeafVector(tl_node_id);
   switch (forest_params.leaf_algo) {
     case leaf_algo_t::CATEGORICAL_LEAF:
       ASSERT(vec.size() == forest_params.num_classes,
              "inconsistent number of classes in treelite leaves");
       fil_node->val.idx = find_class_label_from_one_hot(&vec[0], vec.size());
       break;
+    case leaf_algo_t::VECTOR_LEAF: {
+      ASSERT(vec.size() == forest_params.num_classes,
+             "inconsistent number of classes in treelite leaves");
+      fil_node->val.idx = fil_node_id * forest_params.num_classes;
```

Comment:
could you quickly benchmark to see if doing the multiplication in-kernel can ever slow it down? I mean an edge case. This would support models that take >8GiB due to multiple classes